Increase CP version check accuracy, bump dep for clickstream to 5.0.1

rmoff opened the pull request:
@ybyzek I've tested this but would appreciate a second set of eyes on it please :)

miguno commented on kafka-clients/consumer/pom.xml:
```diff
@@ -35,10 +35,8 @@
-        <kafka.scala.version>2.11</kafka.scala.version>
-        <confluent.version>2.0.1</confluent.version>
-        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <kafka.version>0.10.1.0</kafka.version>
```
We want this to be `0.10.1.0-cp1` I think (we use Confluent artifacts also under `kafka-streams/`).

miguno commented on kafka-clients/producer/pom.xml:
```diff
@@ -35,10 +35,8 @@
-        <kafka.scala.version>2.11</kafka.scala.version>
-        <confluent.version>2.0.1</confluent.version>
-        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <kafka.version>0.10.1.0</kafka.version>
```
Same here: add `-cp1`

miguno commented:
Are you planning any further updates on this @mhowlett ?

ewencp commented:
This should be 3.1.1 now, right?

rmoff committed:
Move datagen startup into separate script
so that it can be invoked on its own for the
KSQL product page tutorial

rmoff committed:
Increase accuracy of Confluent Platform version check

Existing deps on 5.0 will work, but where greater
precision is needed (5.0.1) will handle that too

rmoff committed:
add back check erroneously commented out

rmoff commented:
hmm some other commits bundled in, let me remove those

rmoff committed:
revert d0ade66

ybyzek commented on utils/helper.sh:
```diff
@@ -31,7 +31,8 @@ function check_ccloud() {
 
-  actual_version=$( confluent version | tail -1 | awk -F':' '{print $2;}' | awk '$1 > 0 { print substr($1,1,3)}' )
+  # This returns the actual version running trimmed to the same accuracy as the supplied expected_version
+  actual_version=$( confluent version | tail -1 | awk -F': ' -v exp_ver=$expected_version '{print substr($2,1,length(exp_ver))}')
```
Clever enhancement @rmoff !

ybyzek commented on clickstream/start.sh:
```diff
@@ -4,9 +4,9 @@
-check_running_elasticsearch 5.6.5 || exit 1
-check_running_grafana 5.0.3 || exit 1
+check_running_cp 5.0.1 || exit 
+#check_running_elasticsearch 5.6.5 || exit 1
```
Why is `check_running_elasticsearch` and `check_running_grafana` commented out?

rmoff replied:
by mistake, reverted in https://github.com/confluentinc/examples/pull/85/commits/80292d5a7ae549f9eca39f7b363aaacb2e6da455

ybyzek commented on clickstream/start.sh:
```diff
@@ -4,9 +4,9 @@
-check_running_cp 5.0 || exit 
-check_running_elasticsearch 5.6.5 || exit 1
-check_running_grafana 5.0.3 || exit 1
+check_running_cp 5.0.1 || exit 
```
Is there a reason CP *must* be running 5.0.1?  Will it fail with 5.0.0 or 5.0.2?  If it works with 5.0.0 and 5.0.2, then why not leave as 5.0?

rmoff replied:
there's a bug in RUN SCRIPT which was fixed in 5.0.1, so yes it'll fail in 5.0.0. 
The dep is basically >=5.0.1

ybyzek reviewed (changes requested):
Some questions to discuss before merging.

ybyzek commented:
PR is out of date and has been handled in recent versions.
